VoodooI2CELAN/VoodooI2CELANTouchpadDriver.cpp

```diff
@@ -491,17 +498,32 @@ bool VoodooI2CELANTouchpadDriver::start(IOService* provider) {
     }
     // set interrupts AFTER device is initialised
     interrupt_source = IOInterruptEventSource::interruptEventSource(this, OSMemberFunctionCast(IOInterruptEventAction, this, &VoodooI2CELANTouchpadDriver::interrupt_occurred), api, 0);
+    
     if (!interrupt_source) {
-        IOLog("%s::%s Could not get interrupt event source\n", getName(), elan_name);
-        goto start_exit;
-    }
-    publish_multitouch_interface();
-    if (!init_device()) {
-        IOLog("%s::%s Failed to init device\n", getName(), elan_name);
-        goto start_exit;
+        IOLog("%s::%s Could not get interrupt event source, trying to fallback on polling\n", getName(), elan_name);
+        interrupt_simulator = IOTimerEventSource::timerEventSource(this, OSMemberFunctionCast(IOTimerEventSource::Action, this, &VoodooI2CELANTouchpadDriver::simulateInterrupt));
+        if (!interrupt_simulator) {
+            IOLog("%s::%s Could not get timer event source\n", getName(), elan_name);
+            goto start_exit;
+        }
+        publish_multitouch_interface();
+        if (!init_device()) {
+            IOLog("%s::%s Failed to init device\n", getName(), elan_name);
+            return NULL;
```

Comment:
Don't you want to `goto start_exit` here? Same for L521.

Reply:
Done